index.bs

```diff
@@ -671,43 +751,54 @@ To [=get the next iteration result=] for a {{FileSystemDirectoryHandle}} |handle
 and its async iterator |iterator|:
 
 1. Let |promise| be [=a new promise=].
+1. [=Enqueue the following steps=] to the [=file system queue=]:
+  1. Let |directory| be the result of [=locating an entry=]
+     given |handle|'s [=FileSystemHandle/locator=].
+  1. Let |accessResult| be the result of running |directory|'s
+     [=file system entry/query access=] given "`read`".
 
-1. Let |directory| be the result of [=locating an entry=]
-   given |handle|'s [=FileSystemHandle/locator=].
-1. If |directory| is `null`, [=/reject=] |result| with a
-   "{{NotFoundError}}" {{DOMException}} and abort.
-  1. [=Assert=]: |directory| is a [=directory entry=].
-
-1. Let |access| be the result of running |directory|'s
-   [=file system entry/query access=] given "`read`".
-
-1. If |access| is not "{{PermissionState/granted}}",
-   [=reject=] |promise| with a "{{NotAllowedError}}" {{DOMException}} and
-   return |promise|.
-
-1. Let |child| be a [=/file system entry=] in |directory|'s [=directory entry/children=],
-   such that |child|'s [=file system entry/name=] is not contained in |iterator|'s [=past results=],
-   or `null` if no such entry exists.
-
-   Note: This is intentionally very vague about the iteration order. Different platforms
-   and file systems provide different guarantees about iteration order, and we want it to
-   be possible to efficiently implement this on all platforms. As such no guarantees are given
-   about the exact order in which elements are returned.
-
-1. If |child| is `null`, then:
-  1. [=/Resolve=] |promise| with `undefined`.
-
-1. Otherwise:
-  1. [=set/Append=] |child|'s [=file system entry/name=] to |iterator|'s [=past results=].
-  1. If |child| is a [=file entry=]:
-    1. Let |result| be the result of <a>creating a child `FileSystemFileHandle`</a>
-       with |handle|'s [=FileSystemHandle/locator=] and
-       |child|'s [=file system entry/name=] in |handle|'s [=relevant Realm=].
-  1. Otherwise:
-    1. Let |result| be the result of <a>creating a child `FileSystemDirectoryHandle`</a>
-       with |handle|'s [=FileSystemHandle/locator=] and
-       |child|'s [=file system entry/name=] in |handle|'s [=relevant Realm=].
-  1. [=/Resolve=] |promise| with (|child|'s [=file system entry/name=], |result|).
+  1. [=Queue a storage task=] with |handle|'s [=relevant global object=] to
+     run these steps:
+    1. If |accessResult|'s [=file system access result/permission state=]
+       is not "{{PermissionState/granted}}":
+      1. Set |accessErrorName| to |accessResult|'s
+         [=file system access result/error name=] if it is not
+         « the empty string »; otherwise "{{NotAllowedError}}".
+      1. [=/Reject=] |promise| with an |accessErrorName| {{DOMException}}
+         and return |promise|.
+
+    1. If |directory| is `null`, [=/reject=] |result| with a
+       "{{NotFoundError}}" {{DOMException}} and abort these steps.
```

Comment:
Note: this is the only "behavioral" change in this PR. Presumably the "NotAllowedError" should take precedence over the "NotAllowedError", just like everywhere else